docs/sketch-profiles.md

```diff
@@ -0,0 +1,113 @@
+Arduino CLI provides support for reproducible builds through the use of build profiles.
+
+## Sketch project file `sketch.yaml` and build profiles.
+
+A profile is a complete description of all the resources needed to build a sketch. Profiles are defined in a project
+file called `sketch.yaml`. This file will be in YAML format and it will contain one or more profiles, in particular each
+profile will define:
+
+- The board FQBN
+- The target core platform name and version (with the 3rd party platform index URL if needed)
+- A possible core platform name and version, that is a dependency of the target core platform (with the 3rd party
+  platform index URL if needed)
+- The libraries used in the sketch (including their version)
+
+The format of the file is the following:
+
+```
+profiles:
+  <PROFILE_NAME>:
+    notes: <USER_NOTES>
+    fqbn: <FQBN>
+    platforms:
+      - platform: <PLATFORM> (<PLATFORM_VERSION>)
+        platform_index_url: <3RD_PARTY_PLATFORM_URL>
+      - platform: <PLATFORM_DEPENDENCY> (<PLATFORM_DEPENDENCY_VERSION>)
+        platform_index_url: <3RD_PARTY_PLATFORM_DEPENDENCY_URL>
+    libraries:
+      - <LIB_NAME> (<LIB_VERSION>)
+      - <LIB_NAME> (<LIB_VERSION>)
+      - <LIB_NAME> (<LIB_VERSION>)
+
+...more profiles here...
+```
+
+There is a 'profiles:' section containing all the profiles. Each field is self-explanatory, in particular:
+
+- `<PROFILE_NAME>` is the profile identifier, it’s a user-defined field, and the allowed characters are alphanumerics,
+  underscore `_`, dot `.`, and dash `-`
+- `<PLATFORM>` is the target core platform identifier, for example, `arduino:avr` or `adafruit:samd`
+- `<PLATFORM_VERSION>` is the target core platform version required
+- `<3RD_PARTY_PLATFORM_URL>` is the index URL to download the target core platform (also known as “Additional Boards
+  Manager URLs” in the Arduino IDE). This field can be omitted for the official `arduino:*` platforms.
+- `<PLATFORM_DEPENDENCY>`, `<PLATFORM_DEPENDENCY_VERSION>`, and `<3RD_PARTY_PLATFORM_DEPENDENCY_URL>` contains the same
+  information as `<PLATFORM>`, `<PLATFORM_VERSION>`, and `<3RD_PARTY_PLATFORM_URL>` but for the core platform dependency
+  of the main core platform. These fields are optional.
```

Comment:
Reading with a user's eyes, I'm finding it unclear which fields are required and which are optional.

A reference could be the approach taken in the `library.properties` specification:

https://arduino.github.io/arduino-cli/dev/library-specification/#libraryproperties-file-format